quantum/plugins/ibm/aer/accelerator/aer_accelerator.cpp

```diff
@@ -216,6 +217,49 @@ void AerAccelerator::execute(
   }
 }
 
+void AerAccelerator::apply(std::shared_ptr<AcceleratorBuffer> buffer,
+                           std::shared_ptr<Instruction> inst) {
+  static AER::Statevector::State<QV::QubitVector<double>> stateVec;
+  static auto provider = xacc::getIRProvider("quantum");
+  static AER::RngEngine rng(time(NULL));
+
+  if (!noiseModelObj) {
+    noiseModelObj = std::make_shared<AER::Noise::NoiseModel>(noise_model);
+    stateVec.initialize_qreg(buffer->size());
+  }
+  if (inst->isComposite() || inst->isAnalog()) {
+    xacc::error("Only gates are allowed.");
+  }
+
+  auto tempComp = provider->createComposite("tmp");
+  tempComp->addInstruction(inst);
+  auto qobj_str = xacc_to_qobj->translate(tempComp);
+  auto qObjJson = nlohmann::json::parse(qobj_str)["qObject"];
+  AER::Qobj qobj(qObjJson);
+  assert(qobj.circuits.size() == 1);
+  auto circ = qobj.circuits[0];
+  
+  // Output data container
+  AER::ExperimentData data;
+  data.add_metadata("method", stateVec.name());
+  data.add_metadata("measure_sampling", false);
+  auto noiseCirc = noiseModelObj->sample_noise(circ, rng);
```

Comment:
Does this work if there's no noiseModel?

Reply:
Yes, it does. I added a unit test for this `apply` method and it worked for perfect simulation as well.

Reply:
👍